packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -91,63 +82,59 @@ const ConfirmSignupCode = (_: RouteComponentProps) => {
     'Confirmation code is required'
   );
 
-  // When the user types in the code input field, all banners and tooltips should be cleared
-  // Timeout is added to reduce jankiness, but does not include a smooth hiding effect.
-  useEffect(() => {
-    if (clearMessages) {
-      const timer = setTimeout(() => {
-        setCodeErrorMessage('');
-        setBanner({ type: undefined, children: undefined });
-        setClearMessages(false);
-      }, CLEAR_MESSAGES_TIMEOUT);
-      return () => clearTimeout(timer);
-    }
-    return;
-  }, [clearMessages]);
-
-  // Hide the ResendCode button after too many attempts. Redisplay button after a delay.
-  useEffect(() => {
-    if (resendCodeCount > 3) {
-      const timer = setTimeout(() => {
-        setResendCodeCount(0);
-      }, RESEND_CODE_TIMEOUT);
-      return () => clearTimeout(timer);
-    }
-    return;
-  }, [resendCodeCount]);
-
   async function handleResendCode() {
     try {
       await account.sendVerificationCode();
-      setResendCodeCount(resendCodeCount + 1);
-      setResendStatus(ResendStatus['sent']);
-    } catch (e) {
-      setResendStatus(ResendStatus['error']);
+      // if resending a code is succesful, clear any banner already present on screen
+      if (resendStatus !== ResendStatus['sent']) {
+        setBanner({
+          type: undefined,
+          children: undefined,
+        });
+        setResendStatus(ResendStatus['sent']);
+      }
+    } catch (error) {
+      setResendStatus(ResendStatus['not sent']);
+      const localizedErrorMessage = getLocalizedErrorMessage(
+        ftlMsgResolver,
+        error
+      );
+      setBanner({
+        type: BannerType.error,
+        children: <p>{localizedErrorMessage}</p>,
+      });
     }
   }
 
-  function alertSuccessAndGoForward() {
-    alertBar.success(
-      ftlMsgResolver.getMsg(
-        'confirm-signup-code-success-alert',
-        'Account confirmed successfully'
-      )
-    );
-    // TODO in FXA-6519 redirect elsewhere if relying party
-    navigate('/settings', { replace: true });
-  }
-
   async function verifySession(code: string) {
     logViewEvent(`flow.${viewName}`, 'submit', REACT_ENTRYPOINT);
     try {
-      const newsletterSlugs = location.state?.selectedNewsletterSlugs;
       const hasSelectedNewsletters =
         newsletterSlugs && newsletterSlugs.length > 0;
 
-      const options = hasSelectedNewsletters
-        ? { newsletters: newsletterSlugs }
-        : {};
+      const options = {
+        newsletters: hasSelectedNewsletters ? newsletterSlugs : undefined,
+        scopes: isOAuthIntegration(integration)
+          ? await integration.getPermissions()
+          : undefined,
+        service: isOAuthIntegration(integration)
+          ? integration.getService()
+          : undefined,
```

Comment:
@vpomerleau I think we briefly talked about this - I think you _can_ use `&&` here instead but I don't know if that necessarily feels more clear about what's happening. So long as the endpoint is okay with accepting these values as `undefined`, it should be alright. Otherwise we can modify `options` like:

```
if (hasSelectedNewsletters) { 
  options.newsletters = newsletterSlugs;
}
```
 or we could conditionally include the values if they exist, like:

```
const options = {
  ...(hasSelectedNewsletters ? { newsletters: newsletterSlugs } : {}),
  ...(isOAuthIntegration(integration)
    ? {
        scopes: await integration.getPermissions(),
        service: integration.getService(),
      }
    : {}),
};
```

Reply:
Thanks! I'll clean this up!